store/src/java/com/zimbra/cs/imap/ImapHandler.java

```diff
@@ -2437,43 +2476,45 @@ private String getFolderAttrs(ImapPath path, byte returnOptions, Map<ImapPath, I
             attrs.append(attrs.length() == 0 ? "" : " ").append(children ? "\\HasChildren" : "\\HasNoChildren");
         }
 
-        //partial support for special use RFC 6154; return known folder attrs
-        //we also keep support for non-standard XLIST attributes for legacy clients that may still use them
-        if ((DebugConfig.imapForceSpecialUse || (returnOptions & RETURN_XLIST) != 0) && path.belongsTo(credentials)) {
-            //return deprecated XLIST attrs if requested
-            boolean returnXList = (returnOptions & RETURN_XLIST) != 0;
-            switch (iid.getId()) {
-                case Mailbox.ID_FOLDER_INBOX:
-                    if (returnXList) {
-                        attrs.append(attrs.length() == 0 ? "" : " ").append("\\Inbox");
-                    }
-                    break;
-                case Mailbox.ID_FOLDER_DRAFTS:
-                    attrs.append(attrs.length() == 0 ? "" : " ").append("\\Drafts");
-                    break;
-                case Mailbox.ID_FOLDER_TRASH:
-                    attrs.append(attrs.length() == 0 ? "" : " ").append("\\Trash");
-                    break;
-                case Mailbox.ID_FOLDER_SENT:
-                    attrs.append(attrs.length() == 0 ? "" : " ").append("\\Sent");
-                    break;
-                case Mailbox.ID_FOLDER_SPAM:
-                    attrs.append(attrs.length() == 0 ? "" : " ").append(returnXList ? "\\Spam" : "\\Junk");
-                    break;
-                default:
-                    String query = path.getFolder() instanceof SearchFolderStore ?
-                            ((SearchFolderStore) path.getFolder()).getQuery() : null;
-                    if (query != null) {
-                        if (query.equalsIgnoreCase("is:flagged")) {
-                            attrs.append(attrs.length() == 0 ? "" : " ").append(returnXList ? "\\Starred" : "\\Flagged");
-                        } else if (query.equalsIgnoreCase("is:local")) {
-                            attrs.append(attrs.length() == 0 ? "" : " ").append(returnXList ? "\\AllMail" : "\\All");
+        // don't add folder mapping for mounted folders
+        if (!isMounted) {
+            //partial support for special use RFC 6154; return known folder attrs
+            //we also keep support for non-standard XLIST attributes for legacy clients that may still use them
+            if ((DebugConfig.imapForceSpecialUse || (returnOptions & RETURN_XLIST) != 0) && path.belongsTo(credentials)) {
```

Comment:
why not add isMounted clause to this if
if ((DebugConfig.imapForceSpecialUse || (returnOptions & RETURN_XLIST) != 0) && path.belongsTo(credentials) && !isMounted)

Reply:
Included mounted folder check within in if block